matrix/matrix.py

```diff
@@ -0,0 +1,146 @@
+"""Matrix data type class."""
+
+
+class Matrix:
+    """An immutable object representing a matrix data-type.
+
+    Attributes:
+        _matrix (tuple): contains the data about the matrix. Each line of
+            the matrix is represented by a tuple.
+    note:
+        This implementation only supports squared matrices (n * n).
+    """
+
+    def __init__(self, lines):
+        self._validate(lines)
+        self._matrix = lines
+
+    def _validate(self, data: tuple):
+        """Check if the data given is valid.
+
+        note:
+            Valid data is defined if it contains tuples of the same length
+            (Matrix only supports n * n), items can't be iterable.
```

Comment:
why can't be iterable? it contain tuples, which are iterable

Reply:
Incorrect documentation.